### ISSUE TITLE
Wrong return type for comparisons

### ISSUE THREAD

**Issue by amotzop** [microsoft/pyright#3766]
**Describe the bug**
In some cases the return type of a comparison is different from the return type of the relevant hook function, e.g., `>=` and `__ge__`.

**To Reproduce**
It happened to me when comparing with types annotated as either `Any` or `Unknown`.

**Expected behavior**
I think the return type should always be the same, as there is no way to annotate the actual comparison operation.

**Screenshots or Code**
```python
from typing import Any

class Comparable:
    def __ge__(self, other: Any) -> bool:
        ...

def func(base: Comparable, other: Any):
    reveal_type(base >= other)
    reveal_type(base.__ge__(other))
```

The first type reveal will be `Any` while the second will be `bool`.

**VS Code extension or command-line**
This happens both in Pylance (version 2022.7.3) and in the command-line (version 1.1.264).

**Additional context**
This also happend if `other` was un-annotated in `func`.

**Comment by erictraut** [comment]
Yes, this is the intended behavior. The `>=` operator is not the same as a call to `__ge__`. The operator is a complex composite logic that leverages `__ge__` or `__lt__` depending on the types of the LHS and RHS operators. If one of the two operands is `Any`, a type checker cannot say anything about the result other than that it is `Any`. Mypy evaluates the same result in this case.

I recommend not using `Any` if possible. What you probably mean in this case is `object`. If you replace `Any` with `object` in your sample, you will see the behavior you desire.
